src/plugins/walletConnectToDapps/hooks/useSimulateEvmTransaction.ts

```diff
@@ -68,15 +76,52 @@ export const useSimulateEvmTransaction = ({
               feeData,
             })
         : skipToken,
-    staleTime: 30000,
+    staleTime: 10_000,
+    refetchInterval: 10_000,
     retry: false,
   })
 
+  // For optimistic rollups, get full fee data which includes l1GasLimit
+  // This runs after the Tenderly simulation completes, so we get fully accurate gas sim i.e
+  // l2Fee + (l1GasLimit * l1GasPrice)
+  const feeDataQuery = useQuery({
```

Comment:
This is different than the `gasFeeDataQuery` req above (which is irrelevant to the Tx at hand), this actually gets the `l1gasLimit` for the *akschual* Tx at hand (also `l1GasPrice`, but we already have that from `gasFeeDataQuery`)